### PR TITLE
feat: Add docker image that contains apisix all dependencies

### PR DESCRIPTION
**Pull request by gxthrj**
Related to #2783

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    && bin='#! /usr/local/openresty/luajit/bin/luajit\npackage.path = "/usr/local/apisix/?.lua;" .. package.path' \
+    && sed -i "1s@.*@$bin@" /usr/bin/apisix \
+    && mv /usr/local/apisix/deps/share/lua/5.1/apisix /usr/local/apisix \
+    && apk del .builddeps build-base make unzip
+
+FROM alpine:3.11 AS etcd-stage
+
+WORKDIR /tmp
+
+RUN wget https://github.com/etcd-io/etcd/releases/download/v3.4.14/etcd-v3.4.14-linux-amd64.tar.gz \
```
Comment (spacewander) [inline docker/Dockerfile:26]: Better to put `v3.4.14` into an ARG.

File: README_CN.md [hunk 2]
```diff
@@ -151,6 +151,15 @@ A/B 测试、金丝雀发布(灰度发布)、蓝绿部署、限流限速、抵
 
 ## 立刻开始
 
+### 快速体验
+
+使用docker快速build一个包含所有依赖的 apisix 镜像，一键启动。
+
+```shell
+$ docker build -t apache/apisix:whole -f ./docker/Dockerfile .
+$ docker run -v ./docker/config.yaml:/usr/local/apisix/conf/config.yaml -p 9080:9080 -p 2379:2379 -d apache/apisix:whole
```
Comment (spacewander) [inline README_CN.md:160]: Can we test this build in the CI?

File: docker/Dockerfile (hunk 1 above)
Comment (gxthrj) [reply docker/Dockerfile:26]: ok

File: README_CN.md (hunk 2 above)
Comment (gxthrj) [reply README_CN.md:160]: Yes, will add CI

File: docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,54 @@
+FROM openresty/openresty:alpine-fat AS production-stage
```
Comment (moonming) [inline docker/Dockerfile:1]: is this file copy from apisix-docker repo？

**Comment by moonming** [comment]
I think you should create this PR in apisix-docker repo

File: docker/Dockerfile (hunk 3 above)
Comment (gxthrj) [reply docker/Dockerfile:1]: Yes, copy some section from apisix-docker

**Comment by gxthrj** [comment]
> I think you should create this PR in apisix-docker repo

I think it is a quick install feature from the issue #2783 .

File: docker/Dockerfile (hunk 1 above)
Comment (gxthrj) [reply docker/Dockerfile]: Done

File: README_CN.md (hunk 2 above)
Comment (gxthrj) [reply README_CN.md]: Done

File: README.md [hunk 4]
```diff
@@ -153,6 +153,15 @@ A/B testing, canary release, blue-green deployment, limit rate, defense against
 
 ## Get Started
 
+### Quick start
+
+Use `docker` to quickly build an APISIX image which containing all dependencies, and start it with one click.
```
Comment (spacewander) [inline README.md:158]: `which containing` => `which contains`?

Comment (gxthrj) [reply README.md]: Done

**Comment by gxthrj** [comment]
Close this PR now , have resolved in https://github.com/apache/apisix-docker/pull/99

**Pull request closed without merging**